File: backend/app/schemas/seller.py
from pydantic import BaseModel


class SellerPolicyOut(BaseModel):
    seller_id: str
    negotiation_active: bool
    monthly_negotiation_budget: float
    monthly_negotiation_spent: float
    budget_remaining: float
    min_margin_target: float
    segment_strategy: str

    model_config = {"from_attributes": True}


class SellerPolicyIn(BaseModel):
    negotiation_active: bool | None = None
    monthly_negotiation_budget: float | None = None
    min_margin_target: float | None = None
    segment_strategy: str | None = None


class SellerStatsOut(BaseModel):
    seller_id: str
    seller_name: str
    total_negotiations: int
    successful_negotiations: int
    total_discount_given: float
    estimated_orders_saved: int
    net_roi: float
    total_co2_saved_kg: float
    eco_seller_badge: bool


class SellerDashboardOut(BaseModel):
    policy: SellerPolicyOut
    stats: SellerStatsOut

File: frontend/src/api/seller.ts
import client from './client'

export interface SellerDashboard {
  seller_id: string
  seller_name: string
  eco_seller_badge: boolean
  total_co2_saved_kg: number
  weekly_negotiations_closed: number
  weekly_discount_given: number
  estimated_saved_orders: number
  net_roi: number
}

export interface SellerPolicy {
  monthly_negotiation_budget: number
  monthly_negotiation_spent: number
  min_margin_target: number
}

export const getSellerDashboard = (sellerId: string) =>
  client.get<SellerDashboard>(`/seller/${sellerId}/dashboard`).then((r) => r.data)

export const getSellerPolicy = (sellerId: string) =>
  client.get<SellerPolicy>(`/seller/${sellerId}/policy`).then((r) => r.data)

export const updateSellerPolicy = (sellerId: string, policy: Partial<SellerPolicy>) =>
  client.put(`/seller/${sellerId}/policy`, policy).then((r) => r.data)

export const getSellerStats = (sellerId: string) =>
  client.get(`/seller/${sellerId}/stats`).then((r) => r.data)
